sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ln/LnKey.java

```diff
@@ -33,5 +34,25 @@ public LnKey(TLN tln) {
         this.lnClass = tln.getLnClass().get(0);
         this.prefix = tln.getPrefix();
     }
+
+    public static DataAttributeRef updateDataRef(TAnyLN anyLN, DataAttributeRef dataAttributeRef) {
+        DataAttributeRef filter = DataAttributeRef.copyFrom(dataAttributeRef);
+        if(anyLN.isSetLnType()) filter.setLnType(anyLN.getLnType());
+        if(anyLN instanceof TLN0 ln0){
+            if(ln0.isSetInst()) filter.setLnInst(ln0.getInst());
+            if(ln0.isSetLnClass()) filter.setLnClass(ln0.getLnClass().get(0));
+            filter.setPrefix("");
+        } else {
+            if(((TLN)anyLN).isSetInst()) filter.setLnInst(((TLN)anyLN).getInst());
+            if(((TLN)anyLN).isSetLnClass()) filter.setLnClass(((TLN)anyLN).getLnClass().get(0));
+            if(((TLN)anyLN).isSetPrefix()) {
+                filter.setPrefix(((TLN)anyLN).getPrefix());
```

Comment:
use pattern matching switch instead of casting.
Something like : 
```java
switch (anyLN){
    case TLN0 tln0 -> {
        ...
    }
    case TLN tln -> {
        ...
    }
    default -> throw new RuntimeException("not possible");
}
```